### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
Documentation for terra-icon.

@cerner/terra

### CODE REVIEW

File: docs/terra-icon.md [hunk 1]
```diff
@@ -7,70 +7,92 @@ The terra-icon component is used to visually represent a literal or symbolic obj
...
+
+Priority of sizing icons
+1. React prop width and height.
+2. SVG file width and height.
+3. Default to 1em and will scale with font-size.
+
+
+### Bidirectionality
+
+Some icons will mirror when `dir="rtl"`. The mirror behavior of all icons in the terra-icon repository has been determined already. If an SVG has the css class `is-bidi`, it will mirror when `dir="rtl"`. Nothing needs to be done if you are using an SVG from tera-icon.
```
Comment (bjankord) [inline docs/terra-icon.md]: Some icons will mirror *with* `dir="rtl"` ?

**Comment by zhongyn** [comment]
+1

File: docs/terra-icon.md [hunk 2]
```diff
@@ -7,70 +7,92 @@ The terra-icon component is used to visually represent a literal or symbolic obj
...
   color: #C00;
 }
 ```
 
-```html
-<svg viewBox="0 0 48 48" class="terra-Icon u-text-danger" aria-hidden="true" role="img">
-  <use xlink:href="#cancel"></use>
-</svg>
+```jsx
+<MyIcon class="u-text-danger" />
```
Comment (alex-bezek) [inline docs/terra-icon.md]: Just so I understand, we are cool with exposing our utility css classes, it's all the component css classes we want abstracted via react?

**Comment by alex-bezek** [comment]
+1

**Comment by JakeLaCombe** [comment]
+1

File: docs/terra-icon.md [hunk 3]
```diff
@@ -7,70 +7,92 @@ The terra-icon component is used to visually represent a literal or symbolic obj
...
-
-```html
-<svg viewBox="0 0 48 48" class="terra-Icon" aria-label="Close" role="img">
-  <use xlink:href="#close"></use>
-</svg>
-```
+| Prop       | Type   | Default | Description                                                                                                        |
+|------------|--------|---------|--------------------------------------------------------------------------------------------------------------------|
+| isBidi     | bool   | false   | SVG will mirror when dir="rtl".                                                                                    |
+| children   | node   | null    | JSX children                                                                                                       |
```
Comment (JakeLaCombe) [inline docs/terra-icon.md]: Minor:

Do we have to explicitly call out Children as JSX Children?

**Pull request merged**